### PR TITLE
Add JavaScript Next support

### LINKED ISSUE #45: Add support for ES6 new syntax
Hi,

It'd be awesome if this theme could add support the latest JavaScript syntax, [introduced by ES6](https://github.com/lukehoban/es6features).

This follows the thread started here: https://github.com/Benvie/JavaScriptNext.tmLanguage/issues/75

Thanks!

### PR DESCRIPTION
**Pull request by Snack-X**
Actually it won't close #45. It's just a part of it. Still many issues are left.

I cleaned up existing `.tmTheme` files, them added JSNext support. Check commit message of 10ddbade841551393d17e7e925209981b32986cf for details of clean-up.
Sorry for making big PR.

### CODE REVIEW

**Comment by braver** [comment]
> Sorry for making big PR.

No worries, this is great. I'll look into it and intend to pull later today.

File: Solarized (dark).tmTheme [hunk 1]
```diff
@@ -392,13 +383,13 @@
 		</dict>
 		<dict>
 			<key>name</key>
-			<string>CSS: Selector > [] and non-spec tags</string>
+			<string>CSS: Selector &gt; [] and non-spec tags</string>
```
Comment (braver) [inline Solarized (dark).tmTheme]: Could you revert this line?

File: Solarized (dark).tmTheme [hunk 2]
```diff
@@ -926,32 +917,32 @@
...
-					<string>#2aa198</string>
-				</dict>
+			<dict>
+				<key>foreground</key>
+				<string>#2aa198</string>
+			</dict>
 		</dict>
 		<dict>
-			<key>Name</key>
+			<key>name</key>
```
Comment (braver) [inline Solarized (dark).tmTheme]: Please keep the capital N

File: Solarized (dark).tmTheme [hunk 3]
```diff
@@ -926,32 +917,32 @@
...
-					<string>#6c71c4</string>
-				</dict>
+			<dict>
+				<key>foreground</key>
+				<string>#6c71c4</string>
+			</dict>
 		</dict>
 		<dict>
-			<key>Name</key>
+			<key>name</key>
```
Comment (braver) [inline Solarized (dark).tmTheme]: Here as well

**Comment by braver** [comment]
If you could just look at my corrections and mirror them between dark&light, it all looks dandy to me. 

One question though. Would it be better to scope `variable.other.readwrite` to JS/ES?

**Comment by Snack-X** [comment]
It's done. I hope you like it!
By the way, is there any reason of mixing `name` and `Name`? Only Markdown rules are using `Name`.

**Comment by braver** [comment]
Oh really? Guess if you look back on your own efforts they're always less consistent than you thought...

**Pull request merged**